MonsterTrainModdingAPI/Builders/EnhancerDataBuilder.cs

```diff
@@ -0,0 +1,141 @@
+            this.EnhancerPoolIDs = new List<string>();
+        }
+
+        /// <summary>
+        /// Builds the EnhancerData represented by this builder's parameters
+        /// and registers it and its components with the appropriate managers.
+        /// </summary>
+        /// <returns>The newly registered EnhancerData</returns>
+        public EnhancerData BuildAndRegister()
+        {
+            var enhancerData = this.Build();
+            foreach (var pool in EnhancerPoolIDs)
+            {
+                CustomEnhancerPoolManager.AddEnhancerToPool(enhancerData, pool);
+            }
+            return enhancerData;
+        }
+
+        /// <summary>
+        /// Builds the EnhancerData represented by this builder's parameters.
+        /// </summary>
+        /// <returns>The newly created EnhancerData</returns>
+        public EnhancerData Build()
+        {
+            EnhancerData enhancerData = new EnhancerData();
+            var t = Traverse.Create(enhancerData);
+
+            // Set the name for the unity object
+            enhancerData.name = this.ID;
+
+            // Upgrades are contained within a relic effect - this is mandatory or the game will crash
+            List<RelicEffectData> Effects = new List<RelicEffectData>
+            {
+                new RelicEffectDataBuilder
+                {
+                    relicEffectClassName = "RelicEffectCardUpgrade",
+                    ParamCardUpgradeData = Upgrade.Build(),
+                    ParamCardType = CardType,
+                    ParamCharacterSubtype = "SubtypesData_None",
+                }.Build()
+            };
+            t.Field("effects").SetValue(Effects);
+
+            // Grab the LinkedClass from the ClanID
+            this.LinkedClass = CustomCardManager.SaveManager.GetAllGameData().FindClassData(this.ClanID);
+            t.Field("linkedClass").SetValue(LinkedClass);
+
+            // Take care of the localized strings
+            BuilderUtils.ImportStandardLocalization(this.DescriptionKey, this.Description);
+            t.Field("descriptionKey").SetValue(DescriptionKey);
+
+            BuilderUtils.ImportStandardLocalization(this.NameKey, this.Name);
+            t.Field("nameKey").SetValue(NameKey);
+
+            // Create the icon from the asset path
+            if (this.Icon == null && this.AssetPath != null)
+            {
+                string path = "BepInEx/plugins/" + this.AssetPath;
+                if (File.Exists(path))
+                {
+                    byte[] fileData = File.ReadAllBytes(path);
+                    Texture2D tex = new Texture2D(1, 1);
+                    UnityEngine.ImageConversion.LoadImage(tex, fileData);
+                    this.Icon = Sprite.Create(tex, new Rect(0, 0, tex.width, tex.height), new Vector2(0.5f, 0.5f), 128f);
+                }
+            }
+            t.Field("icon").SetValue(Icon);
+
+            // Rarity
```

Comment:
10/10 comment :D

Reply:
This rarity comment indicates that the rarity field is set with the rarity value, which despite being a rarity should actually occur all the time. ;)